style_guide.md

```diff
@@ -0,0 +1,307 @@
+                days: -20,
+              },
+            },
+            maximum: {
+              value: 'now',
+            },
+          },
+        ],
+        ```
+    
+
+- Region Code
+    - Depending on its value, different structure of sections is generated (e.g. certain questions omitted).
+    - Example of how we determine if a Wales or England description is used:
+      ```
+      local radioOptions = (
+        if std.extVar('region_code') == 'GB-WLS' then 'Wales description'
+        else 'England description';
+      );
+      ```
+
+
+## Blocks
+
+- There are two distinct styles we use with block JSON:
+  - Simple (no variants):
+  
+    ```
+    // Imports
+    
+    {
+      // Block JSON
+    }
+    ```
+
+  - Variants:
+    ```
+    // Imports
+    
+    // Variant helper methods
+    local questionTitle(variantParam) = (
+      // Variables relevant to this method
+      // Return relevant string or JSON object
+    );
+    
+    // Main variant method - 'question' or 'content'
+    local question(variantParam) = {
+      // Variant JSON
+    };
+    
+    {
+      // Block JSON
+    }
+    ```
+
+- Block JSON can be a function if an argument is being passed:
+
+    ```
+    function(census_month_year_date) {
+      type: 'Question',
+      id: 'arrive-in-country',
+      question_variants: [
+        {
+          question: question(isProxy=false),
+          when: [rules.isNotProxy],
+        },
+        {
+          question: question(isProxy=true),
+          when: [rules.isProxy],
+        },
+      ],
+    } 
+    ```
+  
+### Variants
+
+- When the value of a property needs to vary, a method should be defined that returns an appropriate value.
+
+#### Single value variants example
+- The most common approach in census schemas is to create variants based on a value of a single variable:
+    ```
+    {
+      type: 'Question',
+      id: 'confirm_dob',
+      question_variants: [
+        {
+          question: question(isProxy=false),
+          when: [rules.isNotProxy],
+        },
+        {
+          question: question(isProxy=true),
+          when: [rules.isProxy],
+        },
+      ],
+    };
+
+    local question(isProxy) = {
+      title: questionTitle(isProxy)
+      id: 'confirm-date-of-birth',
+      title: title,
+      type: 'General',
+      answers: [
+      {
+        id: 'confirm-date-of-birth-answer',
+        mandatory: true,
+        options: [
+          {
+            label: 'Yes',
+            value: 'Yes',
+           },
+          {
+            label: 'No',
+            value: 'No',
+          },
+        ],
+        type: 'Radio',
+        },
+      ],
+    };
+  
+    local questionTitle(isProxy) = (
+      if isProxy then 
+        {
+          text: 'What was <em>{person_name_possessive}</em> age on their last birthday?',
+          placeholders: [
+            placeholders.personNamePossessive,
+        ]} 
+      else 'What was your age on your last birthday?';
+    );
+    ```
```

Comment:
This should be ordered as we would in a real Jsonnet file:
```suggestion
    ```
    local questionTitle(isProxy) = (
      if isProxy then 
        {
          text: 'What was <em>{person_name_possessive}</em> age on their last birthday?',
          placeholders: [
            placeholders.personNamePossessive,
        ]} 
      else 'What was your age on your last birthday?';
    );

    local question(isProxy) = {
      title: questionTitle(isProxy)
      id: 'confirm-date-of-birth',
      title: title,
      type: 'General',
      answers: [
      {
        id: 'confirm-date-of-birth-answer',
        mandatory: true,
        options: [
          {
            label: 'Yes',
            value: 'Yes',
           },
          {
            label: 'No',
            value: 'No',
          },
        ],
        type: 'Radio',
        },
      ],
    };
    
    {
      type: 'Question',
      id: 'confirm_dob',
      question_variants: [
        {
          question: question(isProxy=false),
          when: [rules.isNotProxy],
        },
        {
          question: question(isProxy=true),
          when: [rules.isProxy],
        },
      ],
    }
    ```
```